### PR TITLE
PFEM adaptive time step + Modality to write  on file has been modified

### PR DESCRIPTION
**Pull request by AFranci**
The file now is not opened and closed several times as in the previous version

### CODE REVIEW

File: applications/PfemFluidDynamicsApplication/python_scripts/remesh_fluid_domains_process.py [hunk 1]
```diff
@@ -164,21 +166,24 @@ def ExecuteInitializeSolutionStep(self):
         currentTime=self.main_model_part.ProcessInfo[KratosMultiphysics.TIME]
         currentStep=self.main_model_part.ProcessInfo[KratosMultiphysics.STEP]                
 
-        if(currentStep > 1):
+        if currentStep >= 2 and self.fileTotalVolume is None:
+            self.fileTotalVolume = open("totalVolumeBeforeMeshing.txt",'w')
```
Comment (josep-m-carbonell) [inline applications/PfemFluidDynamicsApplication/python_scripts/remesh_fluid_domains_process.py:170]: usually  FileName.post.csv is a better extension to know that is is a results file and a type of excel file

**Review by josep-m-carbonell — COMMENTED** [review]
Just some comments. For the output files, the extension  ".post.csv" outputfilename.post.csv   use to define better that it is a results file and a type of list or excel file. The other thing is that the calculation and writing of the volume would be better located as an independent process not inserted and merged inside the remeshing process file. Consider it for future improvements.

**Review by maceligueta — APPROVED** [review]

File: applications/PfemFluidDynamicsApplication/python_scripts/remesh_fluid_domains_process.py (hunk 1 above)
Comment (AFranci) [reply applications/PfemFluidDynamicsApplication/python_scripts/remesh_fluid_domains_process.py:170]: thank you josep maria for the feedback!

File: applications/PfemFluidDynamicsApplication/python_scripts/remesh_fluid_domains_process.py [hunk 2]
```diff
@@ -164,21 +166,24 @@ def ExecuteInitializeSolutionStep(self):
...
+
+        if(currentStep > 1 and self.fileTotalVolume is not None):
             for domain in self.meshing_domains:
                 if(domain.Active()):
                     domain.ComputeAverageMeshParameters()
```
Comment (jcotela) [inline applications/PfemFluidDynamicsApplication/python_scripts/remesh_fluid_domains_process.py:175]: Be aware that this function is now not called unless the output file has been initialized. I don't know if this is a problem for you, but the behavior has changed in relation to the original version.

**Review by jcotela — COMMENTED** [review]
👍 looks good to me

**Pull request merged**